rosidl_generator_mypy/__init__.py

```diff
@@ -165,7 +165,7 @@ def to_type_annotation(
 
         # eg: std_msgs/Header[]
         return Annotation(
-            "typing.Sequence[{}]".format(type_annotation.getter),
+            "typing.MutableSequence[{}]".format(type_annotation.getter),
             "typing.Sequence[{}]".format(type_annotation.setter),
```

Comment:
I quickly checked the rosidl_python implementation.
It seems `array.array` is called for instances other than `array.array` at the setter:

https://github.com/ros2/rosidl_python/blob/3c3cf26aaf66bfda1a0486c07cd0425aab7bff59/rosidl_generator_py/resource/_msg.py.em#L450-L453
https://github.com/ros2/rosidl_python/blob/3c3cf26aaf66bfda1a0486c07cd0425aab7bff59/rosidl_generator_py/resource/_msg.py.em#L604

so even if we pass a tuple to the setter, it will always converted it into an array.
That means, we can safely say it's a MutableSequence 👍 

```py
import array
from std_msgs.msg import Float32MultiArray

msg = Float32MultiArray()
msg.data = (1.0, 2.0, 3.0)
assert isinstance(msg.data, array.array)
```

...and unfortunately `array` module itself doesn't support type annotation (but `typeshed` does, so actually we can write `array[float]`), so `MutableSequence` looks good to me.